src/types/routeMethods.ts

```diff
@@ -58,20 +58,20 @@ export type ExtractRouteMethodParams<T> = T extends RouteMethod<infer Params>
 export type ExtractParamsFromPath<
   TPath extends Route['path']
 > = TPath extends Path
-  ? ExtractParamsFromPathString<UnifyParamEnds<TPath['path']>, TPath['params']>
+  ? TPath['params']
   : TPath extends string
-    ? ExtractParamsFromPathString<UnifyParamEnds<TPath>>
+    ? Path<TPath, {}>['params']
     : never
 
 type ParamEnd = '/'
 
-type UnifyParamEnds<
+export type UnifyParamEnds<
```

Comment:
Exporting because this is used by Path. Wanna organize these a little better the the diff is cleaner if I don't move things around.